api/src/api/users.ts

```diff
@@ -132,4 +133,51 @@ router.delete(
   }),
 );
 
+// Add pitch to a user's claimed pitches
+router.put(
+  '/:userId/claim/:pitchId',
+  errorWrap(async (req: Request, res: Response) => {
+    if (!isValidMongoId(req.params.userId)) {
+      res.status(400).json({
+        success: false,
+        message: 'Bad user ID format',
+      });
+      return;
+    } else if (!isValidMongoId(req.params.pitchId)) {
+      res.status(400).json({
+        success: false,
+        message: 'Bad pitch ID format',
+      });
+      return;
+    }
+
+    const user = await User.findById(req.params.userId);
+    if (!user) {
+      res.status(404).json({
+        success: false,
+        message: 'User not found with id',
+      });
+      return;
+    }
+
+    const pitch = await Pitch.findById(req.params.pitchId);
```

Comment:
Something to consider (not sure if this would make things cleaner or easier but I'm putting this option out here) - mongoose has a `findByIdAndUpdate(id, {$push: {claimedPitches: pitch}}) which is equivalent to line 172 of ur code